domain/consensus/utils/consensushashing/calculate_signature_hash.go

```diff
@@ -0,0 +1,196 @@
+package consensushashing
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/constants"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/hashes"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/serialization"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/subnetworks"
+	"github.com/pkg/errors"
+)
+
+// SigHashType represents hash type bits at the end of a signature.
+type SigHashType uint32
+
+// Hash type bits from the end of a signature.
+const (
+	SigHashAll          SigHashType = 0x1
+	SigHashNone         SigHashType = 0x2
+	SigHashSingle       SigHashType = 0x3
```

Comment:
I think it'll be a tiny bit more clear that this is a bit field if you write it as 0x01.

Reply:
we could even go full literal: `0b00000001` which makes it easier to see the bits and masks with the naked eye (ie `SigHashMask = 0x1f` becomes: `0b00011111`)